src/components/screens/bookmarks/addBookmark/addBookmark.js

```diff
@@ -67,23 +72,36 @@ class AddBookmark extends React.Component {
     }
   }
 
-  updateLabelIfDelegate(prevProps) {
-    const { account } = this.props;
-    const { fields: { label } } = this.state;
-    if (account.data.delegate === prevProps.account.data.delegate) return;
+  updateFields(
+    prevProps,
+    { account, bookmarks, token: { active } },
+    { fields, fields: { label } },
+  ) {
+    if (account.data.address === prevProps.account.data.address) return;
 
-    if (account.data.delegate && account.data.delegate.username !== label.value) {
-      const data = { value: account.data.delegate.username, readonly: true };
-      this.updateField({
-        name: 'label',
-        data,
-      });
-    } else if (label.readonly) {
-      this.updateField({
-        name: 'label',
-        data: { value: '', readonly: false },
-      });
+    const bookmark = bookmarks[active].find(mark => mark.address === account.data.address);
+
+    let newFields = {
+      ...fields,
+      address: { value: account.data.address, readonly: true },
+      label: { value: bookmark ? bookmark.title : '', readonly: false },
+    };
+
+    if (account.data.delegate !== prevProps.account.data.delegate) {
+      if (account.data.delegate && account.data.delegate.username !== label.value) {
+        const data = { value: account.data.delegate.username, readonly: true };
+        newFields = { ...newFields, label: data };
+      } else if (label.readonly) {
+        newFields = { ...newFields, label: { value: '', readonly: false } };
+      }
     }
```

Comment:
This part of the code the overly imperative and the conditional statements don't seem to cover all of the possible branches. Please review.